server/services/ai.ts

```diff
@@ -1,33 +1,185 @@
 import { Game } from "../types/Game";
 import { Player } from "../types/Player";
+import { DiplomaticState } from "../types/Diplomacy";
+import { AiState, KnownAttack } from "../types/Ai";
+import CarrierService from "./carrier";
+import CombatService from "./combat";
+import DistanceService from "./distance";
+import PlayerService from "./player";
+import ShipTransferService from "./shipTransfer";
+import StarService from "./star";
 import StarUpgradeService from "./starUpgrade";
+import TechnologyService from "./technology";
+import WaypointService from "./waypoint";
+import { Star } from "../types/Star";
+import { Carrier } from "../types/Carrier";
+import {getOrInsert, reverseSort, notNull, maxBy} from "../utils";
+import {CarrierWaypoint, CarrierWaypointActionType} from "../types/CarrierWaypoint";
+import ReputationService from "./reputation";
+import DiplomacyService from "./diplomacy";
+import PlayerStatisticsService from "./playerStatistics";
 
 const FIRST_TICK_BULK_UPGRADE_SCI_PERCENTAGE = 20;
 const FIRST_TICK_BULK_UPGRADE_IND_PERCENTAGE = 30;
 const LAST_TICK_BULK_UPGRADE_ECO_PERCENTAGE = 100;
 
-export default class AIService {
+const Heap = require('qheap');
+const mongoose = require("mongoose");
+
+enum AiAction {
```

Comment:
Should we move the enums and interfaces defined in this file to the `Ai.ts` types file?

Reply:
Considering that they're only used internally, I felt they were fine here. They are not exported anyways.